extensions/guacamole-auth-cas/src/main/resources/casConfig.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Config block which registers CAS-specific field types.
+ */
+angular.module('guacCAS').config(['formServiceProvider',
+        function guacCASConfig(formServiceProvider) {
+
+    // Define field for ticket from CAS service
+    formServiceProvider.registerFieldType("GUAC_CAS_TICKET", {
+        templateUrl   : '',
+        controller    : 'guacCASController',
+        module        : 'guacCAS'
+    });
+
+}]);
+
+/**
+ * Config block which augments the existing routing, providing special handling
+ * for the "ticket=" parameter provided by the CAS authentication process.
+ */
+angular.module('index').config(['$routeProvider','$windowProvider',
+        function indexRouteConfig($routeProvider,$windowProvider) {
```

Comment:
Issues with the query parameter vs. Angular's routing notwithstanding, the following issues remain:

1. This is a `.config()` block which doesn't actually do any configuration. Besides being not very Angular-y, this will only run once when the application first loads. Any time after that will not work as expected, including when navigating from page to page or after logging out.
2. As you've noted, it's sloppy.

Have you had any luck using the same technique as my WIP OpenID auth?

https://github.com/mike-jumper/incubator-guacamole-client/blob/9deaac26df8220efeca90a0c9f8e9bdde39c86ff/extensions/guacamole-auth-openid/src/main/resources/openidConfig.js#L34-L52

If not ... there must be some reasonable way of doing this. Let me know if the above doesn't work (and an example of the URL used by CAS for its redirect), and I'll dig in to see if this can be fixed globally somehow.